addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,374 +1,456 @@
+        return super(hr_evaluation, self).write(vals)
+
+    @api.one
+    def update_appraisal_url(self, url, email_to, token):
+        url = urlparse.urlparse(url).path[1:]
+        if token:
+            url = url + '/' + token
+        self.appraisal_url = url
+        self.email_to = email_to
         return True
 
-    def survey_req_done(self, cr, uid, ids, context=None):
-        for id in self.browse(cr, uid, ids, context=context):
-            flag = False
-            wating_id = 0
-            if not id.evaluation_id.id:
-                raise osv.except_osv(_('Warning!'), _("You cannot start evaluation without Appraisal."))
-            records = id.evaluation_id.survey_request_ids
-            for child in records:
-                if child.state == "draft":
-                    wating_id = child.id
-                    continue
-                if child.state != "done":
-                    flag = True
-            if not flag and wating_id:
-                self.survey_req_waiting_answer(cr, uid, [wating_id], context=context)
-        self.write(cr, uid, ids, {'state': 'done'}, context=context)
+    @api.one
+    def create_token(self, email, url, partner_id):
+        """ Create response with token """
+        survey_response_obj = self.env['survey.user_input']
+        token = uuid.uuid4().__str__()
+        survey_response_obj.create({
+            'survey_id': url.id,
+            'deadline': self.date_close,
+            'date_create': datetime.datetime.now().strftime(DF),
+            'type': 'link',
+            'state': 'new',
+            'token': token,
+            'evaluation_id': self.id,
+            'partner_id': partner_id.id if partner_id else None,
+            'email': email})
+        return token
+
+    @api.one
+    def create_receiver_list(self, appraisal_receiver):
+        """ Create one mail by recipients and __URL__ by link with identification token """
+        mail_obj = self.env['mail.thread']
+        mail_template_obj = self.env['email.template']
+        find_no_email = [employee.name for record in appraisal_receiver for employee in record['employee_ids'] if not employee.work_email]
+        if find_no_email:
+            raise Warning(_("Following employees do not have configured an email address. \n- %s") % ('\n- ').join(find_no_email))
+        for record in appraisal_receiver:
+            for emp in record['employee_ids']:
+                email = tools.email_split(emp.work_email) and tools.email_split(emp.work_email)[0] or False
+                if email:
+                    partner_id = mail_obj._find_partner_from_emails(email) or emp.user_id.partner_id or None
+                    token = self.create_token(email, record['survey_id'], partner_id)[0]
+                    self.update_appraisal_url(record['survey_id'].public_url, email, token)
+                    render_template = mail_template_obj.generate_email_batch(self.mail_template.id, [self.id])
```

Comment:
@msh-odoo :  In our case: send survey with unique token  for each employee. if we put this line inside of first loop, It's gives always wrong information  so we need to this line outside of second line.